system-parachains/coretime/coretime-kusama/src/lib.rs

```diff
@@ -0,0 +1,880 @@
+}
+
+parameter_types! {
+	pub const PotId: PalletId = PalletId(*b"PotStake");
+	pub const SessionLength: BlockNumber = 6 * HOURS;
+	/// StakingAdmin pluralistic body.
+	pub const StakingAdminBodyId: BodyId = BodyId::Defense;
+}
+
+/// We allow Root and the `StakingAdmin` to execute privileged collator selection operations.
+pub type CollatorSelectionUpdateOrigin = EitherOfDiverse<
+	EnsureRoot<AccountId>,
+	EnsureXcm<IsVoiceOfBody<GovernanceLocation, StakingAdminBodyId>>,
+>;
+
+impl pallet_collator_selection::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Balances;
+	type UpdateOrigin = CollatorSelectionUpdateOrigin;
+	type PotId = PotId;
+	type MaxCandidates = ConstU32<100>;
+	type MinEligibleCollators = ConstU32<4>;
+	type MaxInvulnerables = ConstU32<20>;
+	// should be a multiple of session or things will get inconsistent
+	type KickThreshold = ConstU32<PERIOD>;
+	type ValidatorId = <Self as frame_system::Config>::AccountId;
+	type ValidatorIdOf = pallet_collator_selection::IdentityCollator;
+	type ValidatorRegistration = Session;
+	type WeightInfo = weights::pallet_collator_selection::WeightInfo<Runtime>;
+}
+
+parameter_types! {
+	/// One storage item; key size is 32; value is size 4+4+16+32 bytes = 56 bytes.
+	pub const DepositBase: Balance = deposit(1, 88);
+	/// Additional storage item size of 32 bytes.
+	pub const DepositFactor: Balance = deposit(0, 32);
+}
+
+impl pallet_multisig::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeCall = RuntimeCall;
+	type Currency = Balances;
+	type DepositBase = DepositBase;
+	type DepositFactor = DepositFactor;
+	type MaxSignatories = ConstU32<100>;
+	type WeightInfo = weights::pallet_multisig::WeightInfo<Runtime>;
+}
+
+impl pallet_utility::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeCall = RuntimeCall;
+	type PalletsOrigin = OriginCaller;
+	type WeightInfo = weights::pallet_utility::WeightInfo<Runtime>;
+}
+
+// Create the runtime by composing the FRAME pallets that were previously configured.
+construct_runtime!(
+	pub enum Runtime
+	{
+		// System support stuff.
+		System: frame_system = 0,
+		ParachainSystem: cumulus_pallet_parachain_system = 1,
+		Timestamp: pallet_timestamp = 3,
+		ParachainInfo: parachain_info = 4,
+
+		// Monetary stuff.
+		Balances: pallet_balances = 10,
+		TransactionPayment: pallet_transaction_payment = 11,
+
+		// Collator support. The order of these 5 are important and shall not change.
+		Authorship: pallet_authorship = 20,
+		CollatorSelection: pallet_collator_selection = 21,
+		Session: pallet_session = 22,
+		Aura: pallet_aura = 23,
+		AuraExt: cumulus_pallet_aura_ext = 24,
+
+		// XCM & related
+		XcmpQueue: cumulus_pallet_xcmp_queue = 30,
+		PolkadotXcm: pallet_xcm = 31,
+		CumulusXcm: cumulus_pallet_xcm = 32,
+		MessageQueue: pallet_message_queue = 34,
+
+		// Handy utilities.
+		Utility: pallet_utility = 40,
+		Multisig: pallet_multisig = 41,
```

Comment:
Worth having a Coretime proxy?

Reply:
I've thought more about this and actually think that the coretime ownership model means that there is no real meaningful scope to give someone on the coretime chain which is useful but not potentially destructive.

Excluding the ones that need sudo or don't check the origin, the broker calls are:
```rust
`purchase`
`assign`
`renew`
`transfer`
`pool`
`partition`
`interlace`
`claim_revenue`
`purchase_credit`
```

Taking the potentially destructive ones first:
- `assign`: the caller specifies the parachain ID.
- `renew`: the caller specifies the core, so any core could be renewed.
- `transfer`: self explanatory.
- `pool`: the caller defines the account that gets paid for the contribution.
- `purchase_credit`: the caller specifies the beneficiary of the credit.
So none of these can be trustlessly delegated.

- `purchase`: establishes the caller as the owner, but if they can't then assign/transfer/pool then there is no point in them owning the region.
- `partition`/`interlace`: can only be done to the caller's owned regions, and even if that was allowed, then we fall back to the argument for purchasing.

Which leaves the only operation that could be trustlessly delegated: `claim_revenue`, but this can come from any signed origin for a small transaction fee.

Reply:
I'll push what I have since it could still be useful for a (probably multisig) proxy to have access to all Broker calls without division.